File: Java/selenium-java/src/test/java/org/example/test_empty_passw.java
package org.example;

import org.openqa.selenium.*;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
public class test_empty_passw {
    private String userVal = global_variable.username_one;
    private String usernameLocator = global_variable.username;
    private String passwordLocator = global_variable.password;
    private String loginLocator = global_variable.login_button;
    private String fail_container = global_variable.fail_container;
    private WebDriver driver;
    private WebDriverWait wait;
    private String cta_text = "Epic sadface: Password is required";

    public test_empty_passw(){
        driver = new FirefoxDriver();
        wait = new WebDriverWait(driver, Duration.ofSeconds(3));
    }

    public static void main(String[] args) {
        test_empty_passw tep = new test_empty_passw();
        tep.empty_passw();
    }

    public Boolean empty_passw(){
        Boolean result = false;
        driver.get("https://saucedemo.com");

        try {
            WebElement uname = wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath(usernameLocator)));
            WebElement login = wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath(loginLocator)));

            String thisUrl = driver.getCurrentUrl();
            uname.sendKeys(userVal);
            login.click();

            WebElement fail_cont = wait.until(ExpectedConditions.presenceOfElementLocated(By.xpath(fail_container)));
            String cont_cta = fail_cont.getAttribute("textContent");
            Boolean assertionResult = cont_cta.equals(cta_text);
            // String res = assertionResult.toString();
            if (assertionResult){
                System.out.println("Fail happen as expected, success!");
                result = true;
            }
            else {
                System.out.println("Fail not happen as expected, failed!");
                result = false;
            }
        } catch (TimeoutException e) {
            System.out.println("Timeout!");
            result = false;
        }
        driver.quit();
        return result;
    }
}

File: Python/LoginPage/test_empty_passw.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.common.exceptions import NoSuchElementException
import sys
sys.path.append('C:/Users/Admin/Documents/selenium/Python')
import global_variable

# IMPORTANT VARIABLE DECLARATION ======================================
driver = webdriver.Firefox()
driver.get("https://www.saucedemo.com/")
wait = WebDriverWait(driver, 5)

username_locator = (By.ID, global_variable.username)
login_button_locator = (By.XPATH, global_variable.login_button)

user_val = global_variable.username_one 
#pass_val = global_variable.password_val

fail_cont = (By.XPATH, global_variable.fail_container)
cta_text = "Epic sadface: Password is required"

count_check = 0
# IMPORTANT VARIABLE DECLARATION ======================================


def input_check(element_input, val):
    global count_check
    count_check += 1
    try:
        attr = wait.until(EC.presence_of_element_located(element_input))

        if attr.is_displayed():
            attr.send_keys(val)
            attr.send_keys(Keys.RETURN)
            print(f"Step [{count_check}] input '{val}' as {element_input} is Succeed ✔")

            exp_url = "https://www.saucedemo.com/"
            url = driver.current_url
            try:
                element = wait.until(EC.presence_of_element_located(login_button_locator))
                
                if element.is_displayed():
                    element.click()
                    
                    try:
                        fail_obj = wait.until(EC.presence_of_element_located(fail_cont))
                        fail_cta = driver.find_element(By.XPATH, global_variable.cta_fail).get_attribute("textContent")
                        #print(fail_cta)
                        assert fail_cta == cta_text, f"Expected is '{cta_text}' while the Actual is '{fail_cta}'"

                    except NoSuchElementException:
                        print("No such element found!")
                        return False
                    
                    except AssertionError:
                        print("Unmatched CTA text")
                        return False
                    
                    else:
                        if fail_obj.is_displayed() and url == exp_url:
                            print(f"Step [2] click and show fail is Succeed ✔")
                            return True
                        else:
                            print(f"Step [2] click and show fail is Failed ✘")
                            print(url)
                            return False
                        
                else:
                    print("Can't handle the process!")
                    return False
                
            except NoSuchElementException:
                print("Element not Found!")
                return False
                
        else:
            print(f"Step [{count_check}] input '{val}' as {element_input} is Failed ✘")
            return False
           
    except Exception as e:
        print(f"Can't handle the process! : {e}")
        return False

def run_test():
    print("Test Started...")
    step_results = False
    
    # ONE STEP CHECK
    print("=======================================================")
    step_results = input_check(username_locator, user_val)

    # Check overall test result
    if(step_results == True):
        print("Testcase - Empty Password, is running successfully!")
        print("=======================================================")
        driver.close()
        return True
    else:
        print('Testcase - Empty Password, is Failed!')
        print("=======================================================")
        driver.close()
        return False

# Call the main function if this script is executed

if __name__ == "__main__":
    run_test()